### ISSUE TITLE
Rucio--what's the right configuration for RSE limits

### ISSUE THREAD

**Issue by StevenCTimm** [DUNE/data-mgmt-ops#344]
A few of our sites have MaxSpaceAvailable set.  Have asked rucio support what's the right combination of limits to set
and where it's documented where they all are.

Steve

(rucio129_python3) [dunepro@dunesl7gpvm01 mcruciosam]$ for rse in `rucio list-rses` ; do rucio-admin rse info $rse | grep -A3 'RSE limits' ; done
RSE limits:
===========
RSE limits:
===========
  MinFreeSpace: 500000000000000 B
  MaxBeingDeletedFiles: 1000 B
RSE limits:
===========
RSE limits:
===========
  MaxSpaceAvailable: 1000000000000000 B
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
  MinFreeSpace: 1000000000000 B
  dunepro: 999000000000000 B
RSE limits:
===========
  MaxSpaceAvailable: 300000000000000 B
  MaxBeingDeletedFiles: 1000 B
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
  MaxBeingDeletedFiles: 100000 B
  MinFreeSpace: 1 B
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
  MinFreeSpace: 100000000000 B
  MaxBeingDeletedFiles: 1000 B
RSE limits:
===========
RSE limits:
===========
  MaxBeingDeletedFiles: 1000 B
  MinFreeSpace: 100000000000 B
RSE limits:
===========
  MaxBeingDeletedFiles: 1000 B
  MaxSpaceAvailable: 1000000000000000 B
RSE limits:
===========
  dunepro: 1000000000000000 B
RSE limits:
===========
  MinFreeSpace: 50000000000000 B
  MaxBeingDeletedFiles: 1000 B
RSE limits:
===========
  MaxBeingDeletedFiles: 100 B
RSE limits:
===========
RSE limits:
===========
  MaxSpaceAvailable: 1000000000000000 B
RSE limits:
===========
RSE limits:
===========
RSE limits:
===========
  MaxSpaceAvailable: 900000000000000 B
RSE limits

**Comment by StevenCTimm** [comment]
Dimitrios says that MinFreeSpace is the only one that matters.. individual account things or MaxSpaceAVailable don't matter.

**Comment by StevenCTimm** [comment]
can try this on SurfSara

**Comment by StevenCTimm** [comment]
So you need MinFreeSpace set plus the following set:

from rucio.client import Client
client = Client()
client.set_rse_usage(rse='DUNE_US_FNAL_DISK_STAGE', source='static', used=-1, free=10 ** 15 - 1)
client.add_rse_attribute(rse='DUNE_US_FNAL_DISK_STAGE', key='source_for_total_space', value='static')
client.add_rse_attribute(rse='DUNE_US_FNAL_DISK_STAGE', key='source_for_used_space', value='rucio')
client.set_rse_limits(rse='DUNE_US_FNAL_DISK_STAGE', name='MinFreeSpace', value=100 * 10 ** 12)

**Comment by StevenCTimm** [comment]
We have 6 of our UK RSE's that have a "computed" value for the usage.
Does anyone remember where that came from and how that is being updated?

Ditto for the "json" usage on those that have it.. what is doing the updates there?
Likewise for the "storage" usage..

There are a couple of our dcache sites at Fermilab that currently only have a "rucio" usage and nothing else.
I need to get something in there for DUNE_US_FNAL_DISK_STAGE so I can get the reaper working properly.

At the moment from what Dmitrios Christidis is telling me, the only reason any deletion is happening on any
of our RSE's is because I have set some of them to GreedyDeletion.. according to him our MinFreeSpace is
configured all wrong.

Steve Timm

**Comment by StevenCTimm** [comment]
Basically before I set  "static" usage records on any of our RSE's I have to understand the provenance of how we got to where we are, where the "computed" and "json"  and "storage" records came from and why FNAL_DCACHE and DUNE_US_FNAL_DISK_STAGE don't have any of them.. .however I can set the MinFreeSpace record on all of them because that is understood.

Also a number of our RSE's have user-flavoured RSE limits.. these appear to do nothing according to Dimitrios, neither does the MaxSpaceAvailable.  

Also the above 6 uses with sourceForTotalSpace and sourceForUsedSpace those settings appear to be ignored,
should be source_for_total_space and source_for_used_space according to Dimitrios.

EDINBURGH
  sourceForTotalSpace: computed
  sourceForUsedSpace: rucio
FNAL_DCACHE
FNAL_DCACHE_PERSISTENT
FNAL_DCACHE_STAGING
  sourceForTotalSpace: computed
  sourceForUsedSpace: rucio
FNAL_DCACHE_TEST
FNAL_SCRATCH_DCACHE
IMPERIAL
LANCASTER
  sourceForTotalSpace: computed
  sourceForUsedSpace: rucio
LIVERPOOL
  sourceForTotalSpace: computed
  sourceForUsedSpace: rucio
MANCHESTER
  sourceForTotalSpace: computed
  sourceForUsedSpace: rucio
